### PR TITLE
[RLlib] Benchmark and regression test yaml cleanup and restructuring.

### PR DESCRIPTION
**Pull request by sven1977**
- `tuned_examples/regression_tests` removed; all tests will be run for both frameworks (`use_pytorch` will be removed from the yamls and replaced by command line overrides in BAZEL py_test). Also, the regression tests will be run as single, independent py_tests for each Algo+Env+Framework to better be able to catch run errors (hard to track right now).

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/tuned_examples/ddpg/halfcheetah_2020-04-26/01_DDPG_HalfCheetah-v2_use_pytorch=True_2020-04-26/config.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+actor_hiddens:
```
Comment (ericl) [inline rllib/tuned_examples/ddpg/halfcheetah_2020-04-26/01_DDPG_HalfCheetah-v2_use_pytorch=True_2020-04-26/config.yaml:1]: Can we keep these yamls runnable with rllib train?

File: rllib/utils/test_utils.py [hunk 2]
```diff
@@ -51,11 +51,11 @@ def framework_iterator(config=None,
             logger.warning("framework_iterator skipping {} (tf not "
                            "installed)!".format(fw))
             continue
-        elif fw == "eager" and not eager_mode:
+        elif fw == "tfe" and not eager_mode:
```
Comment (ericl) [inline rllib/utils/test_utils.py:54]: Could we split this PR into a couple pieces:
1) add the basic infrastructure code and a simple example
2) port rest of examples

File: rllib/tuned_examples/ddpg/halfcheetah_2020-04-26/01_DDPG_HalfCheetah-v2_use_pytorch=True_2020-04-26/config.yaml (hunk 1 above)
Comment (sven1977) [reply rllib/tuned_examples/ddpg/halfcheetah_2020-04-26/01_DDPG_HalfCheetah-v2_use_pytorch=True_2020-04-26/config.yaml]: All these have been removed again and placed into the github.com/ray-project/rl-experiments repo.

File: rllib/utils/test_utils.py (hunk 2 above)
Comment (sven1977) [reply rllib/utils/test_utils.py]: Was not supposed to be here. Reverted this.

Comment (sven1977) [reply rllib/utils/test_utils.py]: Done.

File: rllib/tuned_examples/sac/pendulum-sac.yaml [hunk 3]
```diff
@@ -1,13 +1,16 @@
 # Pendulum SAC can attain -150+ reward in 6-7k
 # Configurations are the similar to original softlearning/sac codebase
-pendulum_sac:
+pendulum-sac:
```
Comment (ericl) [inline rllib/tuned_examples/sac/pendulum-sac.yaml:3]: A lot of these examples are linked from the documentation rst in the algorithms page. Did you make sure none of those links are broken?

File: rllib/tuned_examples/a3c/cartpole-a2c.yaml [hunk 4]
```diff
@@ -1,9 +1,10 @@
-cartpole-a2c-torch:
+cartpole-a2c:
     env: CartPole-v0
     run: A2C
     stop:
         episode_reward_mean: 100
```
Comment (ericl) [inline rllib/tuned_examples/a3c/cartpole-a2c.yaml:5]: Does it actually make sense to move these all into the general set of tuned examples? The problem is that the stop conditions for the regression tests are set quite conservatively.

Comment (sven1977) [reply rllib/tuned_examples/a3c/cartpole-a2c.yaml:5]: Well, they are certainly "tuned" (or should be), so I think that's fine. I agree that we should up the stop-condition to make things harder and more solid. 100 for CartPole is definitely not enough. Let's make all these 150. I think the Pendulum tests are ok.

File: rllib/tuned_examples/sac/pendulum-sac.yaml (hunk 3 above)
Comment (sven1977) [reply rllib/tuned_examples/sac/pendulum-sac.yaml:3]: +1. Checking ...

**Review by ericl — APPROVED** [review]

File: rllib/tuned_examples/sac/pendulum-sac.yaml (hunk 3 above)
Comment (sven1977) [reply rllib/tuned_examples/sac/pendulum-sac.yaml:3]: done

**Pull request merged**